addon/services/infinity.js

```diff
@@ -166,7 +347,7 @@ export default Service.extend({
           this._notifyInfinityModelLoaded();
         } else if (increment == 1) {
           // if list still needs to populate the screen only if we are loading the next page (not previous pages)
-          let infinityLoaderElem = document.querySelector('.infinity-loader');
+          infinityLoaderElem = findElem(infinityLoaderElem);
           if (infinityLoaderElem && this._viewportHeight(infinityModel) > infinityLoaderElem.offsetTop) {
```

Comment:
This reads to me as though the service is now in charge of deciding when the infinityLoader is in viewport. Is that correct?

I thought we had Ember In Viewport in place to do this?

Reply:
So this came from a request to keep loading elements until the loader element was out of the viewport (load 5 items but need to keep loading or else infinity-loader is busted).

I should look into perhaps passing a variable instead indicating if still in viewport when loadNextPage is called.

Reply:
IMO - the infinity-loader should be notified of it's collection changing (we used to use observers for this, but we could subscribe to lifecycle hooks on the infinity model) and when it's collection is updated, the infinity-loader itself should check if it's still in view.